tests/test_file_utils.py

```diff
@@ -15,3 +17,37 @@
 )
 def test_round_up(size, alignment, result):
     assert round_up(size, alignment) == result
+
+
+@pytest.fixture()
+def fake_file() -> io.BytesIO:
+    return io.BytesIO(b"0123456789abcdefghijklmnopqrst")
+
+
+class TestLimitedStartReader:
+    def test_seek_forward(self, fake_file):
+        reader = LimitedStartReader(fake_file, 5)
+        reader.seek(10)
+        assert reader.tell() == 10
+
+    def test_seek_backward(self, fake_file):
+        reader = LimitedStartReader(fake_file, 5)
+        reader.seek(10)
+        reader.seek(-4, io.SEEK_CUR)
+        assert reader.tell() == 6
+
+    def test_seek_before_start(self, fake_file):
+        reader = LimitedStartReader(fake_file, 5)
+        reader.seek(10)
+        reader.seek(-6, io.SEEK_CUR)
+        assert reader.tell() == 5
+
+    def test_seek_to_end_of_file(self, fake_file):
+        reader = LimitedStartReader(fake_file, 5)
+        reader.seek(-1, io.SEEK_END)
+        assert reader.tell() == len(fake_file.getvalue()) - 1
+
+    def test_read_only(self, fake_file):
+        with pytest.raises(TypeError):
+            reader = LimitedStartReader(fake_file, 5)
```

Comment:
This line can be out of the `pytest.raises` context so we would be sure that the `write` function raises the error.

Reply:
Good catch!